### DISCUSSION TITLE
`spin-trigger` dependencies for custom triggers

### DISCUSSION THREAD

**Discussion by smndtrl** [spinframework/spin#2336] [General]
Hej,

I'm working on a custom trigger and was wondering if there's a way to make it a bit leaner to create them. I noticed that a build of my action ran for 15+ minutes and was pulling in e.g. `rustformers/llm` despite my trigger having only a simple network socket and some byte parsing.

Is this because even a custom trigger would include all of `spin` including the custom worlds and basically producing a standalone runner?

**Comment by lann** [comment]
> Is this because even a custom trigger would include all of spin including the custom worlds and basically producing a standalone runner?

Pretty much, yes. While it is technically possible to build a Spin runtime that doesn't include all of these features it currently is quite difficult and requires duplicating large portions of the standard spin-trigger base code. This is something I very much want to improve on but it will probably require redesigning some of Spin's core architecture which will be a major project.

**Comment by smndtrl** [comment]
Thank you for confirming!

I love the flexibility that is already there as part of custom triggers and how easy it makes building on top of it. Extending with custom worlds, remote sources, caching, etc. From learning about the trigger plugins to having it running my custom DAG of components in 3 days.

Keep up the great work
